### PR TITLE
Add Ledger hardware wallet support

### LINKED ISSUE #717: Add ledger hardware wallet support
Would be cool to support this hardware wallet:
https://www.ledgerwallet.com

Currently blocked by #3249

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes #717 and #4593 

UI/UX wise it's almost identical to the Trezor implementation, with the main difference that instead of connecting to the device via popup (Trezor connect) I've implemented a 2 way communication through an iframe in the background page, and the iframe interacts to the device sending messages back and forth to the extension.  More info [here](https://github.com/brunobar79/eth-ledger-bridge-keyring)

Another important thing is that we're supporting 2 different HD paths for ledger.  
They have updated the previous one (which MEW / MyCrypto still uses) so by supporting both we're allowing users to use their legacy accounts too.

 @cjeria @bdresser please take a look at the design and provide feedback: 

ledger

video: http://recordit.co/zVcKUIPUYY

### CODE REVIEW

**Comment by bdresser** [comment]
Awesome work @brunobar79 this is good stuff. Glad we're supporting both HD paths, that will save support a giant headache :) 

UI looks smooth, only small suggestions:
* We could swap the two helper paragraphs below HD path: start with "These are the accounts available..." then have "If you don't see your existing Ledger address..." It reads a little smoother that way, but I also realize the instructions are closest to the element they're talking about... your call.
* Rather than two full-width buttons (Connect with Trezor / Connect with Ledger) it might look nice with two side-by-side, I'll let @cjeria make the call on this one. 
* When you successfully create a new account, is the default title `LEDGER #1` ? Could we drop the `#` since other accounts are just numbered (Account 1, Account 2). Sorry my OCD made me say something 😬

Want me to reach out to Ledger about an affiliate link to add in with the Trezor one?

**Comment by brunobar79** [comment]
@bdresser  That's great feedback!

RE: order of the paragraphs, I've played around and all the combinations had pro and cons like you said.  I don't wanna say here's the list of your accounts and then show something else. Maybe we can move the paragraph "If you don't see your existing Ledger address(es) ..." under the account list.
 connect. WDYT?

I like the side by side idea for the buttons. We could use the original Ledger and Trezor logos and remove the words "Connect to"

RE: Affiliate link, do it!

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -119,8 +119,8 @@
   "close": {
     "message": "Close"
   },
-  "chromeRequiredForTrezor":{
-    "message": "You need to use MetaMask on Google Chrome in order to connect to your TREZOR device."
+  "chromeRequiredForHardwareWallets":{
+    "message": "You need to use Metamask on Google Chrome in order to connect to your Hardware Wallet."
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: I think this should be cased differently: MetaMask

**Review by whymarrh — DISMISSED** [review]

**Comment by cjeria** [comment]
The first connect screen is looking busier now with the added connect button. I agree with @bdresser maybe place the button side by side and add a very clear connect button like this:

image

On the Unlock screen, here's what I suggest we do:
image


Also remove the "#" symbol from the account names.

**Review by danfinlay — DISMISSED** [review]
Reviewed as a team, looks good!

**Review by danfinlay — APPROVED** [review]

**Comment by MicahZoltu** [comment]
Is this chrome only, or also FF and Edge?

**Comment by gre** [comment]
great work!

**Comment by gnarco** [comment]
Hi

I just bought a second ledger to to some test and connect it to Metamask.

Work great until I request on a dApps to sign a transaction.
It goes to metamask ask for confirmation, wait, and then the transaction appear in a "wait list".

I'm not on the main net but on a custom one. Perhaps it's the issue ? 

Also, perhaps I'm in the wrong place for chatting about this, don't hesitate to redirect me :)

**Comment by mtsalenc** [comment]
Looks like this is still a thing and may not get fixed:
https://www.reddit.com/r/ledgerwallet/comments/gcdx1f/any_hope_for_eip712_support/

**Review by Ktt155 — COMMENTED** [review]
Should i creatva ledger wallet ?

**Pull request merged**